src/repositories/application-repository.js

```diff
@@ -162,42 +204,59 @@ export const searchApplications = async (
   limit = 10,
   sort = { field: 'createdAt', direction: 'DESC' }
 ) => {
-  // TODO 1182 impl
-  const applicationsDB = await db
+  const query = buildSearchQuery(searchText, searchType, filter)
+
+  const total = await db
     .collection(APPLICATION_COLLECTION)
-    .find({}, { projection: { _id: 0 } })
-    .toArray()
-  // let query = buildSearchQuery(searchText, searchType, filter)
-
-  const total = applicationsDB.length
-  const applications = applicationsDB.map((app) => {
-    return { ...app, status: { status: 'agreed' }, flags: [] }
-  })
-  const applicationStatus = applicationsDB.map((app) => {
-    return { status: 'agreed', total }
-  })
-
-  // total = await models.application.count(query)
-
-  // if (total > 0) {
-  //   applicationStatus = await models.application.findAll({
-  //     attributes: [
-  //       'status.status',
-  //       [sequelize.fn('COUNT', 'application.id'), 'total']
-  //     ],
-  //     ...query,
-  //     group: ['status.status', 'flags.id'],
-  //     raw: true
-  //   })
-  //   sort = evalSortField(sort)
-  //   query = {
-  //     ...query,
-  //     order: [sort],
-  //     limit,
-  //     offset
-  //   }
-  //   applications = await models.application.findAll(query)
-  // }
+    .countDocuments(query)
+
+  let applicationStatus = []
```

Comment:
What is this bit doing? We're getting all the applications which match the query, that makes sense, but why do we need an array of applicationStatuses?

Reply:
I would say can we check to see if its needed, as I cant really figure out why it would be. If backoffice needs those statuses, it can iterate over the array of applications presumably?

Reply:
ah makes sense